pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -273,6 +275,56 @@ def pattern_matching(circuit_dag, pattern_dag):
 
     Returns:
         list(Match): the list of maximal matches.
+
+    **Example**
+
+    First let's consider the following circuit
+
+    .. code-block:: python
+
+        def circuit():
+            qml.S(wires=0)
+            qml.PauliZ(wires=0)
+            qml.S(wires=1)
+            qml.CZ(wires=[0, 1])
+            qml.S(wires=1)
+            qml.S(wires=2)
+            qml.CZ(wires=[1, 2])
+            qml.S(wires=2)
+            return qml.expval(qml.PauliX(wires=0))
+
+    where we want to find all maximal matches of a pattern containing a sequence of two ``pennylane.S`` gates and
+    a ``pennylane.PauliZ`` gate:
+
+    .. code-block:: python
+
+        with qml.tape.QuantumTape() as pattern:
+            qml.S(wires=0)
+            qml.S(wires=0)
+            qml.PauliZ(wires=0)
+
+    >>> circuit_dag = qml.commutation_dag(circuit)()
+    >>> pattern_dag = qml.commutation_dag(pattern)()
+    >>> all_max_matches = qml.pattern_matching(circuit_dag, pattern_dag)
+
+    It is possible to access the matches by looping through the list. The first integers indices represent the gates
+    in the pattern and the second intergers the gates in the circuit (by order of appearance).
```

Comment:
```suggestion
    The matches are accessible by looping through the list outputted by `qml.pattern_matching`. This output is a list of two lists containing indices. The first list indexes the gates
    of the pattern and the second list provides indices for the gates in the circuit (by order of appearance).
```

Reply:
Also, a specific example explaining the indices and the gates in the pattern and the circuit would be great for the readers.